src/model.rs

```diff
@@ -86,7 +86,7 @@ impl Cmd for PrepareEx {
         let &PrepareEx(ref ex) = self;
         let ex = ex::Experiment::load(&ex.0)?;
         // Shared experiment prep
-        ex::fetch_gh_mirrors(&ex)?;
+        ex.fetch_repo_crates()?;
         ex::capture_shas(&ex)?;
         ex::download_crates(&ex)?;
         ex::frob_tomls(&ex)?;
```

Comment:
If all of the functions here, for example, were switched to use futures and chained together, with the implementations just wrapped with `futures::lazy`, that would be a reasonable sized change, I think.